BUILD

```diff
@@ -39,6 +39,23 @@ java_library(
     ],
 )
 
+load("//tools:jarjar.bzl", "jarjar_library")
+
+genrule(
+    name = "rules_file",
+    outs = ["rules_file.txt"],
+    cmd = "echo \"rule com.google.auto.common.** dagger.shaded.auto.common.@1\" > $@",
+)
+
+jarjar_library(
+    name = "shaded_compiler",
+    rules_file = ":rules_file.txt",
+    deps = [
+        "//compiler",
+        "@com_google_auto_auto_common//jar",
```

Comment:
Apart from avoiding bugs with class loaded by name, none (well, when your goal is to distribute a standalone app as a single jar, there's no value in relocating, but that's not the case here).

I consider that a bug: those deps should have been relocated, or not shaded.